opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -435,9 +435,11 @@ mod tests {
             })
             .expect("callback registration should succeed");
 
+        _ = meter_provider.force_flush();
```

Comment:
why `force_flush` here?

Reply:
Just to trigger the [callback code](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/metrics/periodic_reader.rs#L434) deterministically. This lets us assert things on the receiver end with certainty instead of waiting for `interval * 2` milliseconds and hoping that the sender sends the data in that time.